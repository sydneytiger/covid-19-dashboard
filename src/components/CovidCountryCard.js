import React from 'react'

function CovidCountryCard({data}) {

  return (
    <div>
      <div className="covid-country-card">
        <div className="header">
          <img src={data.countryInfo.flag} alt={data.country} />
          <h3>{data.country}</h3>
        </div>
        <div className="body">
          <h4>Overall</h4>
          <div>Cases: <span>{data.cases}</span></div>
          <div>Active: <span>{data.active}</span></div>
          <div>Critical: <span>{data.critical}</span></div>
          <div>Deaths: <span>{data.deaths}</span></div>
          <div>Tests: <span>{data.tests}</span></div>

          <h4>Today</h4>
          <div>Cases: <span>{data.todayCases}</span></div>
          <div>Deaths: <span>{data.todayDeaths}</span></div>
        </div>
      </div>
    </div>
  )
}

export default CovidCountryCard
