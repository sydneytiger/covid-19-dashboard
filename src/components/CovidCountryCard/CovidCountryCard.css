.container {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 20px;
  margin: auto;
  width: 940px;
}

.covid-country-card {
  border-radius: 10px;
  box-shadow: 0 2px 20px rgba(0,0,0,0.1);
  overflow: hidden;
  width: 260px;
  background-color: #fff;
  padding: 30px
}

.covid-country-card .header {
  display: flex;
  flex-direction: row;
  text-transform: uppercase;
}

.covid-country-card .header img {
  width: 60px;
  height: 60px;
  border-radius: 50%;
  object-fit: fill;
  margin-right: 20px;
  box-shadow: 0 2px 20px rgba(0,0,0,0.3);
}

.covid-country-card .body h4 {
  margin-bottom: 10px;
}

.covid-country-card .body span{
  font-weight: bold;
  margin-left: 10px;
}


@media (max-width: 940px) {
  .container {
    grid-template-columns: 1fr;
    justify-items: center;
  }
}