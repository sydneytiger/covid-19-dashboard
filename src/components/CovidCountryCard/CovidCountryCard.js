import React from 'react';
import './CovidCountryCard.css';

function CovidCountryCard({data}) {

  return (
    <div>
      <div className="covid-country-card">
        <div className="header">
          <img src={data.flag} alt={data.country} />
          <h3>{data.country}</h3>
        </div>
        <div className="body">
          <h4>Overall</h4>
          <div>Cases: <span className="overall-cases">{data.cases.toLocaleString()}</span></div>
          <div>Active: <span className="overall-active">{data.active.toLocaleString()}</span></div>
          <div>Critical: <span className="overall-critical">{data.critical.toLocaleString()}</span></div>
          <div>Deaths: <span className="overall-deaths">{data.deaths.toLocaleString()}</span></div>
          <div>Tests: <span className="overall-tests">{data.tests.toLocaleString()}</span></div>

          <h4>Today</h4>
          <div>Cases: <span className="today-cases">{data.todayCases.toLocaleString()}</span></div>
          <div>Deaths: <span className="today-deaths">{data.todayDeaths.toLocaleString()}</span></div>
        </div>
      </div>
    </div>
  )
}

export default React.memo(CovidCountryCard)
