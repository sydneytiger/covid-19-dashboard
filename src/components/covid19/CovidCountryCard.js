import React from 'react';
import '../../style/CovidCountryCard.css';

function CovidCountryCard({data}) {

  return (
    <div>
      <div className="covid-country-card">
        <div className="header">
          <img src={data.countryInfo.flag} alt={data.country} />
          <h3>{data.country}</h3>
        </div>
        <div className="body">
          <h4>Overall</h4>
          <div>Cases: <span>{data.cases.toLocaleString()}</span></div>
          <div>Active: <span>{data.active.toLocaleString()}</span></div>
          <div>Critical: <span>{data.critical.toLocaleString()}</span></div>
          <div>Deaths: <span>{data.deaths.toLocaleString()}</span></div>
          <div>Tests: <span>{data.tests.toLocaleString()}</span></div>

          <h4>Today</h4>
          <div>Cases: <span>{data.todayCases.toLocaleString()}</span></div>
          <div>Deaths: <span>{data.todayDeaths.toLocaleString()}</span></div>
        </div>
      </div>
    </div>
  )
}

export default React.memo(CovidCountryCard)
