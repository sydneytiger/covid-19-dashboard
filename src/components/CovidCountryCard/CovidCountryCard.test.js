/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import CovidCountryCard from './CovidCountryCard';

const testData = {
  updated:1594205402819,
  country:"Australia",
  iso2:"AU",
  iso3:"AUS",
  flag:"https://disease.sh/assets/img/flags/au.png",
  cases:8886,
  todayCases:131,
  deaths:106,
  todayDeaths:0,
  recovered:7487,
  todayRecovered:32,
  active:1296,
  critical:8,
  tests:2853342,
  population:25504621
}

  // the very first time .toMatchSnapshot run, it generates a .snap file in __snapshots__ folder
  // the .snap file will be the source of true that everytime this test run against to.
  // If you make any change e.g. change div tag to p tag in source code. This test is going to fail
  // If it is what you want, press 'u' key in test console to regenerate snapshot 
it('should render (with enyzme shallow)', () => {
  const wrapper = shallow(<CovidCountryCard data={testData} />);
  expect(wrapper).toMatchSnapshot();
})

// similar to shallow, render generate the snapshot in details as any custom components indside
// the wrapped component will be rendered.
// e.g. <Parent /> component contains <Child /> component. shallow outputs <Child /> in snapshot
// whereas render outputs whatever the HTML elements in <Child /> component in snapshot. 
it('should render (with enzyme render)', () => {
  const wrapper = render(<CovidCountryCard data={testData} />);
  expect(wrapper).toMatchSnapshot();
})

describe('Test CovideCountryCard', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<CovidCountryCard data={testData} />);
  });

  it('should display the country name correctly', () => {
    const countryName = wrapper.find('h3').text();
    expect(countryName).toEqual('Australia');;
  })

  it('should display the country flag', () => {
    const countryFlag = wrapper.find('img').prop('src');
    expect(countryFlag).toEqual('https://disease.sh/assets/img/flags/au.png');
  })

  it('should display the correct overall cases', () => {
    const overallCases = wrapper.find('span.overall-cases').text();
    expect(overallCases).toEqual('8,886');
  })

  it('should display the correct overall active', () => {
    const overallActive = wrapper.find('span.overall-active').text();
    expect(overallActive).toEqual('1,296');
  })

  it('should display the correct overall critical', () => {
    const overallCritical = wrapper.find('span.overall-critical').text();
    expect(overallCritical).toEqual('8');
  })

  it('should display the correct overall deaths', () => {
    const overallDeaths = wrapper.find('span.overall-deaths').text();
    expect(overallDeaths).toEqual('106');
  })

  it('should display the correct overall test', () => {
    const overallTests = wrapper.find('span.overall-tests').text();
    expect(overallTests).toEqual('2,853,342');
  })

  it('should display the correct today deaths', () => {
    const todayDeaths = wrapper.find('span.today-deaths').text();
    expect(todayDeaths).toEqual('0');
  })

  it('should display the correct today cases', () => {
    const todayCases = wrapper.find('span.today-cases').text();
    expect(todayCases).toEqual('131');
  })

});
