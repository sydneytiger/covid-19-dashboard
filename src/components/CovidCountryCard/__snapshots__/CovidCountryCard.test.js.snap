// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render (with enyzme shallow) 1`] = `
<div>
  <div
    className="covid-country-card"
  >
    <div
      className="header"
    >
      <img
        alt="Australia"
        src="https://disease.sh/assets/img/flags/au.png"
      />
      <h3>
        Australia
      </h3>
    </div>
    <div
      className="body"
    >
      <h4>
        Overall
      </h4>
      <div>
        Cases: 
        <span
          className="overall-cases"
        >
          8,886
        </span>
      </div>
      <div>
        Active: 
        <span
          className="overall-active"
        >
          1,296
        </span>
      </div>
      <div>
        Critical: 
        <span
          className="overall-critical"
        >
          8
        </span>
      </div>
      <div>
        Deaths: 
        <span
          className="overall-deaths"
        >
          106
        </span>
      </div>
      <div>
        Tests: 
        <span
          className="overall-tests"
        >
          2,853,342
        </span>
      </div>
      <h4>
        Today
      </h4>
      <div>
        Cases: 
        <span
          className="today-cases"
        >
          131
        </span>
      </div>
      <div>
        Deaths: 
        <span
          className="today-deaths"
        >
          0
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render (with enzyme render) 1`] = `
<div>
  <div
    class="covid-country-card"
  >
    <div
      class="header"
    >
      <img
        alt="Australia"
        src="https://disease.sh/assets/img/flags/au.png"
      />
      <h3>
        Australia
      </h3>
    </div>
    <div
      class="body"
    >
      <h4>
        Overall
      </h4>
      <div>
        Cases: 
        <span
          class="overall-cases"
        >
          8,886
        </span>
      </div>
      <div>
        Active: 
        <span
          class="overall-active"
        >
          1,296
        </span>
      </div>
      <div>
        Critical: 
        <span
          class="overall-critical"
        >
          8
        </span>
      </div>
      <div>
        Deaths: 
        <span
          class="overall-deaths"
        >
          106
        </span>
      </div>
      <div>
        Tests: 
        <span
          class="overall-tests"
        >
          2,853,342
        </span>
      </div>
      <h4>
        Today
      </h4>
      <div>
        Cases: 
        <span
          class="today-cases"
        >
          131
        </span>
      </div>
      <div>
        Deaths: 
        <span
          class="today-deaths"
        >
          0
        </span>
      </div>
    </div>
  </div>
</div>
`;
